src/core/vector/qgsvectorlayerutils.cpp

```diff
@@ -120,6 +120,52 @@ QList<QVariant> QgsVectorLayerUtils::getValues( const QgsVectorLayer *layer, con
   return values;
 }
 
+QList<QVariant> QgsVectorLayerUtils::getUniqueValues( const QgsVectorLayer *layer, const QString &fieldOrExpression, bool &ok, bool selectedOnly, QgsFeedback *feedback )
+{
+  QList<QVariant> uniqueValues;
+  QgsFeatureIterator fit = getValuesIterator( layer, fieldOrExpression, ok, selectedOnly );
+  if ( ok )
+  {
+    std::unique_ptr<QgsExpression> expression;
+    QgsExpressionContext context;
+
+    int attrNum = layer->fields().lookupField( fieldOrExpression );
+    if ( attrNum == -1 )
+    {
+      // use expression, already validated in the getValuesIterator() function
+      expression.reset( new QgsExpression( fieldOrExpression ) );
+      context.appendScopes( QgsExpressionContextUtils::globalProjectLayerScopes( layer ) );
+    }
+
+    QgsFeature feature;
+    while ( fit.nextFeature( feature ) )
+    {
+      QVariant newValue;
+      if ( expression )
+      {
+        context.setFeature( feature );
+        newValue = expression->evaluate( &context );
+      }
+      else
+      {
+        newValue = feature.attribute( attrNum );
+      }
+
+      if ( !uniqueValues.contains( newValue ) )
```

Comment:
if the number of unique values was huge (worst case: each feature has a unique value), then we will have O(n^2) performance if n is the number of features.  We should rather use a QSet<>, at least internally in the method, to have O(n log(n)) worst case performance. It might also be worth adding a limit parameter, defaulting to -1 = unlimited.

The documentation mentions that null values will be omitted from the result set, but this isn't obvious to me from the code.